sympy/matrices/matrices.py

```diff
@@ -2400,7 +2399,8 @@ def dot(self, b):
                 raise TypeError(
                     "`b` must be an ordered iterable or Matrix, not %s." %
                     type(b))
-
+        if (b.cols !=1 and b.rows != 1):
+            raise ShapeError("`b` must has only one row or one column")
```

Comment:
has -> have

Reply:
You should also check for the shape of `self`, and some of below codes will become redundant. And since it was a documented behavior, I think deprecation is necessary and you should update the documentation.